mod.ts

```diff
@@ -108,30 +114,31 @@ export class HMAC {
 
 /** Returns a HMAC of the given msg and key using the indicated hash. */
 export function hmac(
-  hash: string,
+  hash: HashType,
   key: string | Uint8Array,
   msg?: string | Uint8Array,
   inputEncoding?: string,
   outputEncoding?: string
 ): string | Uint8Array {
-  if (SHA1_REGEX.test(hash)) {
-    return new HMAC(new SHA1())
-      .init(key, inputEncoding)
-      .update(msg, inputEncoding)
-      .digest(outputEncoding);
-  } else if (SHA256_REGEX.test(hash)) {
-    return new HMAC(new SHA256())
-      .init(key, inputEncoding)
-      .update(msg, inputEncoding)
-      .digest(outputEncoding);
-  } else if (SHA512_REGEX.test(hash)) {
-    return new HMAC(new SHA512())
-      .init(key, inputEncoding)
-      .update(msg, inputEncoding)
-      .digest(outputEncoding);
-  } else {
-    throw new TypeError(
-      `Unsupported hash ${hash}. Must be one of SHA(1|256|512).`
-    );
+  switch(hash) {
+    case HashType.SHA1:
+      return new HMAC(new SHA1())
+        .init(key, inputEncoding)
+        .update(msg, inputEncoding)
+        .digest(outputEncoding);
+    case HashType.SHA256:
+      return new HMAC(new SHA256())
+        .init(key, inputEncoding)
+        .update(msg, inputEncoding)
+        .digest(outputEncoding);
+    case HashType.SHA512:
+      return new HMAC(new SHA512())
+        .init(key, inputEncoding)
+        .update(msg, inputEncoding)
+        .digest(outputEncoding);
+    default:
+      throw new TypeError(
+        `Unsupported hash ${hash}. Must be one of SHA(1|256|512).`
```

Comment:
and once its a `const enum` please map `hash` to a string here

Reply:
I will do it, but Typescript will prevent this default case (because TSC will throw an error if argument is not a Hash value). So, may be it's not necessary. :)